docs/debug-port.md

```diff
@@ -0,0 +1,108 @@
+# `cloud-hypervisor` debug IO port
+
+`cloud-hypervisor` uses the [`0x80`](https://www.intel.com/content/www/us/en/support/articles/000005500/boards-and-kits.html)
+I/O port to trace user defined guest events.
+
+Whenever the guest write one byte between `0x0` and `0xF` on this particular I/O port,
+`cloud-hypervisor` will log and timestamp that event at the `debug` log level.
+
+It is up to the guest stack to decide when and what to write to the 0x80 port
+in order to signal the host about specific events and have `cloud-hypervisor`
+log it.
+
+There are 3 pre-defined debug port codes. When logging a write of one of those
+codes to the debug port, `cloud-hypervisor` will add a pre-defined string to the
+logs.
+
+| Code  | Event                    | Log string         |
+| ----- | ------------------------ | ------------------ |
+| `0x0` | Firmware starting point  | `Firmware START`   |
+| `0x1` | Kernel starting point    | `Kernel START`     |
```

Comment:
Can we actually allocate ranges please instead. Maybe we could assign the first nibble for each type: e.g. 0x0 to 0x1f firmware 0x20 to 0x3f bootloader (this will be different in EFI case) 0x40 to 0xf5 kernel 0x60 to 0x7f userspace